pam/integration-tests/integration_test.go

```diff
@@ -0,0 +1,112 @@
+package main_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/testutils"
+)
+
+var daemonPath string
+
+func TestIntegration(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	_, stopped := testutils.RunDaemon(ctx, t, daemonPath, testutils.WithSocketPath("/tmp/authd.sock"))
+	t.Cleanup(func() {
+		cancel()
+		<-stopped
+	})
+
+	tests := map[string]struct {
+		tape string
+	}{
+		"Authenticate user successfully":               {tape: "simple_auth"},
+		"Authenticate user with mfa":                   {tape: "mfa_auth"},
+		"Authenticate user with form mode with button": {tape: "form_with_button"},
+		"Authenticate user and reset password":         {tape: "mandatory_password_reset"},
+		"Authenticate user and offer password reset":   {tape: "optional_password_reset"},
+		"Authenticate user switching auth mode":        {tape: "switch_auth_mode"},
+		"Remember last successful auth mode":           {tape: "remember_auth_mode"},
+		"Exit authd if local broker is selected":       {tape: "local_broker"},
+		"Deny authentication if max attempts reached":  {tape: "max_attempts"},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			// #nosec:G204 - we control the command arguments in tests
+			cmd := exec.Command("vhs", filepath.Join("testdata", "tapes", tc.tape+".tape"))
+			cmd.Env = testutils.AppendCovEnv(os.Environ())
+
+			out, err := cmd.CombinedOutput()
+			require.NoError(t, err, "Failed to run tape %q: %v: %s", tc.tape, err, out)
+			t.Cleanup(func() {
+				_ = os.Remove(tc.tape + ".txt")
+				_ = os.Remove(tc.tape + ".gif")
```

Comment:
I don't think these should be deleted on failure, but instead we should save them as artifact so that can be downloaded from CI (or locally) and inspected.

Reply:
Is it possible to handle this?